### PR TITLE
Add inter-bot final bot

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
And the final of the inter-bot @amuramoto

### CODE REVIEW

**Comment by amuramoto** [comment]
@ChrisChinchilla Please change the project name in package.json

**Comment by amuramoto** [comment]
@ChrisChinchilla Both your readmes list this as starter project, but it is the final. Please update.

**Comment by ChrisChinchilla** [comment]
@amuramoto In this particular case I was waiting for the tutorial to be live, the read me will need final updating when the tutorial is live.

**Comment by amuramoto** [comment]
@ChrisChinchilla - Some issues with your PR:

- README gives inaccurate info on how to run the demo. There are various values like page token that need to be set before `npm run start` can be run succeessfully.
- app.js incorrect reference to project and link in the copyright notice

**Comment by ChrisChinchilla** [comment]
@amuramoto 

* 👍 
* Updated, apart from waiting for live tutorial link

Will also apply to `starter` branch.

File: inter-page/marketing-bot/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Inter-bot Tutorial - Final Project for Marketing bot
+
+This is a final project for the marketing bot of the [Messenger Platform inter-bot tutorial](#).
```
Comment (amuramoto) [inline inter-page/marketing-bot/README.md]: update link

File: inter-page/marketing-bot/app.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ * Starter Project for Marketing bot of the Messenger Platform Inter-bot Tutorial
+ *
+ * Use this project as the starting point for following the tutorial.
+ *
+ * TODO: Waiting for link
```
Comment (amuramoto) [inline inter-page/marketing-bot/app.js]: link

File: inter-page/marketing-bot/app.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ */
+
+'use strict';
+
+// Imports dependencies and set up http server
+const
+    request = require('request'),
+    express = require('express'),
+    body_parser = require('body-parser'),
+    ACCESS_TOKEN = process.env.ACCESS_TOKEN,
```
Comment (amuramoto) [inline inter-page/marketing-bot/app.js]: standardize. Capitalize constants or don't

File: inter-page/marketing-bot/app.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        res.status(200).send('EVENT_RECEIVED');
+
+    } else {
+        // Return a '404 Not Found' if event is not from a page subscription
+        res.sendStatus(404);
+    }
+
+});
+
+function processComments(comment) {
```
Comment (amuramoto) [inline inter-page/marketing-bot/app.js]: add description of stub

File: inter-page/support-bot/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Inter-bot Tutorial - Final Project for Support bot
+
+This is a final project for the support bot of the [Messenger Platform inter-bot tutorial](#).
```
Comment (amuramoto) [inline inter-page/support-bot/README.md]: link

File: inter-page/support-bot/app.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+app.post('/webhook', (req, res) => {
+
+    // Parse the request body from the POST
+    let body = req.body;
+
+    // Check the webhook event is from a Page subscription
+    if (body.object === 'page') {
+
+        body.entry.forEach(function (entry) {
```
Comment (amuramoto) [inline inter-page/support-bot/app.js]: convert to arrow function

File: inter-page/marketing-bot/app.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+// Sets server port and logs message on success
+app.listen(process.env.PORT || 5000, () => console.log('webhook is listening'));
+
+app.post('/webhook', (req, res) => {
+    // Parse the request body from the POST
+    let body = req.body;
+
+    // Check the webhook event is from a Page subscription
+    if (body.object === 'page') {
+        body.entry.forEach(function (entry) {
```
Comment (amuramoto) [inline inter-page/marketing-bot/app.js]: convert to arrow function

File: inter-page/support-bot/app.js [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ * Starter Project for Support bot of the Messenger Platform Inter-bot Tutorial
+ *
+ * Use this project as the starting point for following the tutorial.
+ *
+ * TODO: Waiting for link
```
Comment (amuramoto) [inline inter-page/support-bot/app.js]: link

File: inter-page/support-bot/app.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+        }
+    }
+});
+
+
+function handleMessage(sender_psid, received_message) {
+    let response;
+
+    if (received_message.text) {
+        switch (received_message.text.replace(/[^\w\s]/gi, '').trim().toLowerCase()) {
```
Comment (amuramoto) [inline inter-page/support-bot/app.js]: is this switch statement needed? Looks like it only has a default

File: inter-page/support-bot/app.js [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+            default:
+                response = {
+                    "text": `You sent the message: "${received_message.text}".`
+                };
+                break;
```
Comment (amuramoto) [inline inter-page/support-bot/app.js]: don't need to break on default.

File: inter-page/support-bot/app.js (hunk 9 above)
Comment (ChrisChinchilla) [reply inter-page/support-bot/app.js]: This was more so people could add to it in the future, stay or go?

**Comment by ChrisChinchilla** [comment]
@amuramoto Done

File: inter-page/support-bot/app.js (hunk 9 above)
Comment (amuramoto) [reply inter-page/support-bot/app.js]: @ChrisChinchilla I'd say take it out

File: inter-page/support-bot/app.js (hunk 10 above)
Comment (amuramoto) [reply inter-page/support-bot/app.js]: @ChrisChinchilla This still needs update

File: inter-page/marketing-bot/app.js (hunk 4 above)
Comment (amuramoto) [reply inter-page/marketing-bot/app.js]: @ChrisChinchilla This still needs update

Comment (amuramoto) [reply inter-page/marketing-bot/app.js]: @ChrisChinchilla Sorry, copied 'stub' from the starter. Still needs a description tho.

File: inter-page/support-bot/app.js (hunk 6 above)
Comment (amuramoto) [reply inter-page/support-bot/app.js]: @ChrisChinchilla

**Pull request merged**